game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProSortMoveOptionsUtils.java

```diff
@@ -65,39 +71,23 @@ private ProSortMoveOptionsUtils() {}
         new ArrayList<>(unitAttackOptions.entrySet());
     list.sort(
         (o1, o2) -> {
+          final Collection<Territory> territories1 =
+              filterOutWinningTerritories(o1.getValue(), player, attackMap, calc);
+          final Collection<Territory> territories2 =
+              filterOutWinningTerritories(o2.getValue(), player, attackMap, calc);
```

Comment:
Haven't checked if this works nicely in every case, but have you considered using the Comparator builder methods like
[`Comparator#comparing`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Comparator.html#comparing(java.util.function.Function))?
We have several examples in the code where `Comparator#thenComparing` is used to chain several conditions together.
Depending on the complexity it will either make the code half the size because you only have to provide the key function once instead of twice, or make it close to unreadable because the pattern doesn't match the needs

Reply:
Yeah that might be even nicer. Can look at that as a follow up PR potentially.

Reply:
👍